XenModel/Actions/HealthCheck/XenServerHealthCheckUpload.cs

```diff
@@ -80,34 +80,41 @@ public XenServerHealthCheckUpload(string token, int verbosity, string uploadUrl,
         }
 
         // Request an upload and fetch the upload id from CIS.
-        public string InitiateUpload(string fileName, long size, string caseNumber, System.Threading.CancellationToken cancel)
+        private string InitiateUpload(string fileName, long size, string caseNumber, System.Threading.CancellationToken cancel)
         {
             // Request a new bundle upload to CIS server.
             string url = string.Format(INITIATE_UPLOAD_STRING, uploadUrl, size, fileName.UrlEncode());
             if (!string.IsNullOrEmpty(caseNumber))
                 url += "&sr=" + caseNumber;
             log.InfoFormat("InitiateUpload, UPLOAD_URL: {0}", url);
             
-            // Add the log verbosity level in json cotent.
+            // Add the log verbosity level in json content.
             string verbosity = "{\"verbosity\":\""+ verbosityLevel + "\"}";
             
             try
             {
-                var httpContent = new StringContent(verbosity);
-                httpContent.Headers.ContentType = new MediaTypeHeaderValue("application/json");
-                var response = httpClient.PostAsync(new Uri(url), httpContent, cancel).Result;
-                response.EnsureSuccessStatusCode();
-                var respString = response.Content.ReadAsStringAsync().Result;
-
-                // Get the upload uuid
-                var res = (Dictionary<string, object>)serializer.DeserializeObject(respString);
-                if (res.ContainsKey("id"))
-                    return (string) res["id"];
+                
+                using (var httpContent = new StringContent(verbosity))
+                {
+                    httpContent.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    using (var response = httpClient.PostAsync(new Uri(url), httpContent, cancel).Result)
+                    {
+                        if (response.IsSuccessStatusCode)
+                        {
+                            var respString = response.Content.ReadAsStringAsync().Result;
+
+                            // Get the upload uuid
+                            var res = (Dictionary<string, object>) serializer.DeserializeObject(respString);
+                            if (res.ContainsKey("id"))
+                                return (string) res["id"];
+                        }
+                        log.ErrorFormat("Exception while initiating a new CIS upload. The exception was: {0} ({1})", response.StatusCode, response.ReasonPhrase);

```

Comment:
Strictly speaking this is not an exception. I'd rephrase the message to `Failed to initiate a CIS upload. POST request returned {0} ({1)}`.
Also, not sure if it is possible, but if the status code is 200 and the response does not contain the id, the error will still be logged, which is misleading. We can either put the log in an else block, or preferably (to match the upload chunk case) do `return res.ContainsKey("id") ? (string) res["id"] : "";`